### PR TITLE
Follower replication

### PR DESCRIPTION
**Pull request by jeroiraz**
This PR implements embedded replication.

It's possible to create databases as read-only replicas from other databases.

### CODE REVIEW

File: pkg/replication/replicator.go [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+	}
+
+	txr.mainContext, txr.cancelFunc = context.WithCancel(context.Background())
+
+	txr.nextTx = st.TxId + 1
+
+	txr.running = true
+
+	go func() {
+		for {
```
Comment (byo) [inline pkg/replication/replicator.go:99]: Shouldn't we also check if the context was not cancelled?

```suggestion
		for txr.mainContext.Err() == nil {
```

I smell some subtle timing issues that could cause reopening of the client connection after the `Stop` was called

Comment (byo) [reply pkg/replication/replicator.go:105]: A corner case is if there's some connect / fetchTx error and both the timeout and context cancellation fire at the same time. Then even though context is cancelled, the second path (timeout) can be selected and will cause one more reconnect.

A very rare case thus I don't think it should block this PR.

**Review by byo — APPROVED** [review]
Approved with some minor comment

**Review by byo — APPROVED** [review]
Some tests are failing but looks like the feature is working as expected

**Comment by elimisteve** [comment]
Are there any docs showing us how to use this replication feature? Thanks!

**Comment by jeroiraz** [comment]
> Are there any docs showing us how to use this replication feature? Thanks!

Hi @elimisteve, we described it here: https://docs.immudb.io/master/operations/replication.html

**Comment by elimisteve** [comment]
Thanks @jeroiraz!

**Pull request merged**